### PR TITLE
add API to set access token and add user agent

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

Update API to set access token in instruction view and add `User-Agent` to fetch `Junction` images. (Fixes [3225](https://github.com/mapbox/mapbox-navigation-android/issues/3225) and [3226](https://github.com/mapbox/mapbox-navigation-android/issues/3226))
Refers: https://github.com/mapbox/1tap-android/issues/381

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Update API to set access token in instruction view and add `User-Agent` to fetch `Junction` images.

### Implementation

Update API to set access token in instruction view and add `User-Agent` to fetch `Junction` images.

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR
- [x] We might need to update / push `api/current.txt` files after running `$> make 1.0-core-update-api` (Core) / `$> make 1.0-ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

### CODE REVIEW

**Comment by abhishek1508** [comment]
This PR cannot be tested until https://github.com/mapbox/valhalla-internal/issues/812 has been fixed and merged.
cc @LukasPaczos @Guardiola31337 @zugaldia

**Comment by abhishek1508** [comment]
Tested the above changes with 2 different routes

**Route 1**
origin: lon - 139.7772481; lat - 35.6818019
destination: lon - 139.7756523; lat - 35.6789722

**Route 2**
origin: lon - 139.7374577; lat - 35.6507027
destination: lon - 139.7435047; lat - 35.6549106

The `.gif` below is obtained using **Route1**

ezgif com-optimize

cc @LukasPaczos @Guardiola31337 @danpaz @zugaldia

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java [hunk 1]
```diff
@@ -465,8 +461,9 @@ public void stopNavigation() {
    *
    * @param onNavigationReadyCallback to be set to this view
    */
```
Comment (JunDai) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java:463]: we should update the doc as well.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java [hunk 2]
```diff
@@ -487,7 +484,8 @@ public void initialize(OnNavigationReadyCallback onNavigationReadyCallback) {
    * @param initialMapCameraPosition to be shown once the map is ready
```
Comment (JunDai) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java:484]: same as above

**Comment by abhishek1508** [comment]
For the Beta next week, we can release with the designs we have and revisit it for GA. 
cc @Guardiola31337 @noracalabrese @avi-c @d-prukop @map-angela

**Comment by noracalabrese** [comment]
I agree @Guardiola31337 it does look very small. @map-angela can you share the mockups you had worked on before for junction view in mobile?

**Comment by map-angela** [comment]
I made these awhile ago. 

JunctionView_Android_Horizontal_AT_01
JunctionView_Mobile_03

File: examples/src/main/java/com/mapbox/navigation/examples/ui/BuildingFootprintHighlightActivityKt.kt [hunk 3]
```diff
@@ -47,7 +48,11 @@ class BuildingFootprintHighlightActivityKt : AppCompatActivity(), OnNavigationRe
         setContentView(R.layout.activity_final_destination_arrival_building_highlight)
 
         navigationView.onCreate(savedInstanceState)
-        navigationView.initialize(this, getInitialCameraPosition())
+        navigationView.initialize(
+            this,
+            getInitialCameraPosition(),
+            Utils.getMapboxAccessToken(this)
```
Comment (kmadsen) [inline examples/src/main/java/com/mapbox/navigation/examples/ui/BuildingFootprintHighlightActivityKt.kt:54]: did we really need to introduce this access token requirement here? it's required in a few places and it breaks APIs to change it

cc: @abhishek1508 @Guardiola31337 @LukasPaczos

**Pull request merged**